### PR TITLE
docs: make kata-deploy more visible

### PR DESCRIPTION
**Pull request by fidencio**
This PR moves kata deploy installation instructions to docs/install
folder, update wording and document format, adds an index to general
installation documentat

Fixes: #2450

Signed-off-by: João Vanzuita <joao.vanzuita@de.bosch.com>
Signed-off-by: Fabiano Fidêncio <fabiano.fidencio@intel.com>

### CODE REVIEW

**Comment by fidencio** [comment]
@converge, I've taken the freedom to take https://github.com/kata-containers/kata-containers/pull/2457 and re-open it here, going through (some of) @jodh-intel comments.  Of course, your authorship is maintained.

@jodh-intel, may I bother you for a second review on this one?  I'd love to get it merged sooner than later.

File: docs/install/README.md [hunk 1]
```diff
@@ -12,7 +12,7 @@ Containers.
 
 Packaged installation methods uses your distribution's native package format (such as RPM or DEB).
 
-*Note:* We encourage installation methods that provides automatic updates, it ensures security updates and bug fixes are
+**Note:** We encourage installation methods that provides automatic updates, it ensures security updates and bug fixes are
```
Comment (jodh-intel) [inline docs/install/README.md:15]: Thanks - https://github.com/kata-containers/kata-containers/blob/main/docs/Documentation-Requirements.md#notes ftw! ;)

**Review by jodh-intel — COMMENTED** [review]
Thanks @fidencio - This is something I keep meaning to do and never quite getting around to ;)

As an additional change (and separate commit), it would be great if you could do "something" with the "Packaged installation methods" title. The problem with that section is that it contains the table. But the table lists packaged *and* unpackaged installation methods (such as "manual", "build from source" and arguably "automatic" since no debs/rpms or snaps are used for those options).

**Comment by fidencio** [comment]
This is a test comment, for a different matter. :-)

**Comment by elianerpereira** [comment]
> /test_kata_deploy

Just helping @fidencio to run a quick test in here, sorry for the noise.

**Comment by fidencio** [comment]
@jodh-intel, please, could you take another look at this PR?  I've made some changes in order to make this PR as minimal as possible.

Also, James and I had a discussion in private about, maybe, moving some items out of the `docs/install` part and let move them to the developer contributing guide.  That will be addressed as part of a different PR.

**Review by jodh-intel — APPROVED** [review]
Thanks @fidencio!

lgtm

**Review by wainersm — APPROVED** [review]
This whole PR is a good idea. It looks good to me either. Thanks!

**Comment by converge** [comment]
sorry, I didnt have the time to continue.

**Pull request merged**